File: iosApp/iosApp/Infrastructure/Storages/WeaponConstantData/WeaponDataSource.swift
//
//  WeaponDataSource.swift
//  iosApp
//
//  Created by ウルトラ深瀬 on 2026/05/31.
//

import Foundation
import SharedLogic

final class WeaponDataSource: WeaponDataSourceInterface {
    let list: [Weapon] = [
        .init(id: 0, name: "M1911", capacity: 7, imageName: "pistol"),
        .init(id: 1, name: "RPG-7", capacity: 1, imageName: "rocket-launcher")
    ]
}

File: androidApp/src/main/kotlin/org/example/project/infrastructure/storages/weaponConstantData/WeaponDataSource.kt
package org.example.project.infrastructure.storages.weaponConstantData

import org.example.project.data.dataSourceInterfaces.WeaponDataSourceInterface
import org.example.project.domain.entities.Weapon

class WeaponDataSource: WeaponDataSourceInterface {
    override val list: List<Weapon> = listOf(
        Weapon(
            id = 0,
            name = "M1911",
            capacity = 7,
            imageName = "pistol",
        ),
        Weapon(
            id = 1,
            name = "RPG-7",
            capacity = 1,
            imageName = "rocket_launcher",
        ),
    )
}